src/rez/cli/_util.py

```diff
@@ -60,6 +60,52 @@
 }
 
 
+def load_plugin_cmd():
+    """Load extension plugin subcommand
+
+    The extension plugin module must have attribute `command_behavior`, and
+    the value must be a dict. For example:
+
+        # in your extension plugin module
+        command_behavior = {
+            "hidden": False,   # optional: bool
+            "arg_mode": None,  # optional: None, "passthrough", "grouped"
+        }
+
+    """
+    from rez.config import config
+    from rez.utils.logging_ import print_debug
+    from rez.plugin_managers import plugin_manager
+
+    ext_plugins = dict()
+
+    for plugin_name in plugin_manager.get_plugins("extension"):
+        module = plugin_manager.get_plugin_module("extension", plugin_name)
+
+        if hasattr(module, "command_behavior"):
+            try:
+                data = module.command_behavior.copy()
+                data.update({"module_name": module.__name__})
+                ext_plugins[plugin_name] = data
+
+            except Exception:
+                if config.debug("plugins"):
+                    import traceback
+                    from rez.vendor.six.six import StringIO
+                    out = StringIO()
+                    traceback.print_exc(file=out)
+                    print_debug(out.getvalue())
+
+        elif config.debug("plugins"):
```

Comment:
I would default command_behaviour to something sensible rather than fail to load. Less barrier to entry, and most extensions are only going to require standard (non grouped) arg behaviour.

Reply:
Ah, the extension type plugin can provide more than just sub-command, like forwarding functions, plugin configs, so it was meant to be like this to enable sub-command specifically by providing attribute `command_behaviour`. Since it would not make sense to register a sub-command when the plugin only want to register forwarding functions.

Reply:
implemented in 8d0af73 (see this [comment](https://github.com/nerdvegas/rez/pull/1040#issuecomment-804600475))